node/src/effect.rs

```diff
@@ -974,6 +974,26 @@ impl<REv> EffectBuilder<REv> {
             .await
     }
 
+    pub(crate) async fn get_activation_point_era_id(self) -> Option<u64>
+    where
+        REv: From<ChainspecLoaderRequest> + From<StorageRequest>,
+    {
+        let CurrentRunInfo {
+            activation_point, ..
+        } = self.get_current_run_info().await;
+        info!(?activation_point, "get_activation_point_era_id");
+        match activation_point {
+            ActivationPoint::EraId(era_id) => self
+                .get_key_block_header_for_era_id_from_storage(era_id)
+                .await
+                .map(|block_header| block_header.height()),
+            ActivationPoint::Genesis(_timestamp) => {
+                // Nothing to do at the genesis
+                None
+            }
+        }
```

Comment:
Seems like the height of the first block of the current protocol version is something which could be established in the `ChainspecLoader` and cached there, avoiding repeated calls to storage.

That way, it would be set from when the `ChainspecLoader` is finished initializing and this method doesn't need to return an `Option`.

Reply:
Seems like we do already cache this information at the request, and the indexing happens at the start of the process. Here's the retrieval under the hood https://github.com/casper-network/casper-node/blob/4c520a64b8b5bf20c19cf94b7b66b4110938ac30/node/src/components/storage.rs#L1062-L1074

Reply:
My bad, I misread this method and I was assuming it's caching the header, not the hash. I added a cache for switch block height by era id in 04de5f0a6c0c73e3994d115e9066f9086da0634a